app/components/transUnit/TransUnitService.js

```diff
@@ -170,7 +198,26 @@
        updateSaveButton);
 
     function modifyTranslationText(phrase, newText) {
-      phrase.newTranslation = newText;
+      if (phrase.plural) {
+        var transUnitCtrl = controllerList[phrase.id];
+        phrase.newTranslations[transUnitCtrl.focusedTranslationIndex] = newText;
+      } else {
+        phrase.newTranslation = newText;
+      }
+      EventService.emitEvent(EventService.EVENT.TRANSLATION_TEXT_MODIFIED,
+        phrase);
+      EventService.emitEvent(EventService.EVENT.FOCUS_TRANSLATION,
+        phrase);
+    }
+
+    function modifyTranslationsText(phrase, newTexts) {
```

Comment:
If we treate everything as plurals, I think we would still weant this function, and we could name it `setAllTranslations` or similar.